src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CaseManagementCategoryAppenderTest.java

```diff
@@ -0,0 +1,130 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.times;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event.START_APPEAL;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage.ABOUT_TO_SUBMIT;
+
+import java.util.Arrays;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.*;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+
+@ExtendWith(MockitoExtension.class)
+class CaseManagementCategoryAppenderTest {
+
+    @Mock
+    private Callback<AsylumCase> callback;
+    @Mock
+    private CaseDetails<AsylumCase> caseDetails;
+    @Mock
+    private AsylumCase asylumCase;
+
+    private CaseManagementCategoryAppender caseManagementCategoryAppender;
+    private final Value value = new Value("refusalOfEu","Refusal of application under the EEA regulations");
+    private final DynamicList dynamicList = new DynamicList(value, Arrays.asList(value));
+
+    @BeforeEach
+    public void setUp() {
+        caseManagementCategoryAppender = new CaseManagementCategoryAppender();
+    }
+
+    @Test
+    void handler_checks_case_management_category_value() {
+
+        when(callback.getEvent()).thenReturn(START_APPEAL);
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        Mockito.when(asylumCase.read(APPEAL_TYPE,AppealType.class)).thenReturn(Optional.of(AppealType.EA));
+
+        PreSubmitCallbackResponse<AsylumCase> response =
+            caseManagementCategoryAppender.handle(ABOUT_TO_SUBMIT, callback);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getData()).isNotEmpty();
+        assertThat(response.getData()).isEqualTo(asylumCase);
+        assertThat(response.getErrors()).isEmpty();
+
+
+        Mockito.verify(asylumCase, times(1)).write(CASE_MANAGEMENT_CATEGORY, dynamicList);
+    }
+
```

Comment:
should the test also check the other possible values as per AC? I.e.
refusalofHumanRights
refusalOfEU
deprivation
revocationOfProtection

Reply:
test improved